pkg/controllers/uiplugin/components.go

```diff
@@ -49,12 +52,52 @@ func newServiceAccount(info UIPluginInfo, namespace string) *corev1.ServiceAccou
 	}
 }
 
-func newRole(info UIPluginInfo) *rbacv1.Role {
-	return info.Role
+func newRole(info UIPluginInfo, roleName string) *rbacv1.Role {
+	if info.Role != nil {
+		return info.Role
+	}
+	return &rbacv1.Role{
```

Comment:
maybe we don't need this as we would be creating a role with no rules.

Reply:
I noticed that the code using `NewOptionalUpdater` does not work during my work on the logging-view-plugin as well, because it operates on wrong assumptions: the `NewOptionalUpdater` function does not work with a `nil` resource passed to it, so the current code will crash, when used with a plugin without a role or rolebinding.

This code here effectively creates an empty Role (or RoleBinding below) just so that the "optional updater" can remove the resource should it be present in the cluster.

I'm not sure if we need this complexity here though, because the role/rolebinding resources for the current set of plugins are either always or never created, depending on the plugin type. That's why I removed the `NewOptionalUpdater` in [my version of the fix](https://github.com/xperimental/rhobs-observability-operator/commit/73b2e65d79726c003f4ec86769db419a72a0e088) completely.

Reply:
I had taken a look at your PR and I think you have laid the structure out better than I had in some places. I'm in the process of adding the distributed tracing plugin to this PR as well, but in the process I'm going to be moving the definitions out of the builder and into separate files and making the Role and RoleBinding optional like you have. I'll try to keep the naming and formatting the same as yours to keep any conflicts to a minimum

Reply:
I was tempted to restructure some of the existing code during the implementation for the logging-view-plugin as well, but I refrained from it for now. Getting it to work was the first priority. But I might follow-up with some of the ideas in another PR, if I still remember them... :slightly_smiling_face: